clang-tools-extra/clang-tidy/readability/RedundantMemberInitCheck.cpp

```diff
@@ -18,52 +19,80 @@ using namespace clang::tidy::matchers;
 
 namespace clang::tidy::readability {
 
+static SourceRange
+getFullInitRangeInclWhitespaces(SourceRange Range, const SourceManager &SM,
+                                const LangOptions &LangOpts) {
+  const Token PrevToken =
+      utils::lexer::getPreviousToken(Range.getBegin(), SM, LangOpts, false);
+  if (PrevToken.is(tok::unknown))
+    return Range;
+
+  if (PrevToken.isNot(tok::equal))
+    return {PrevToken.getEndLoc(), Range.getEnd()};
```

Comment:
Is it fine to merge different base case into one? would be more concise and potentially easier to understand.

Reply:
So I was just curious if its fine to merge the above two if conditions into a single one, and then based on ternary operator returning the Range? If you think that's not a good idea for readability purpose then it's totally fine.

Reply:
No it would look ugly:
```
if (PrevToken.is(tok::unknown) || PrevToken.isNot(tok::equal))
   return {(PrevToken.is(tok::unknown) ? Range.getBegin() : PrevToken.getEndLoc()), Range.getEnd()};
```

It would require duplicating an condition. and those 2 conditions are not +- same.
First one is: we get invalid token, then let just return what we can (error case), second we found something else than =, so lets get that range.

Reply:
Yeah that's how I was thinking. Thanks for explaining!